### ISSUE TITLE
Indicator stops working after screen lock

### ISSUE THREAD

**Issue by x1unix** [ubuntu/gnome-shell-extension-appindicator#170]
## Description

AppIndicator icons disappear after screen was locked (and unlocked).
Issue can be fixed by uninstalling the extension, restarting Gnome and installing back the extension.

This issue makes the extension unusable.

## Environment

* **OS**: Ubuntu 18.10
* **GNOME**: 3.30.2
* **Ext version**: latest from `extensions.gnome.org`

## Steps to reproduce

* Install extension from [extensions.gnome.org](https://extensions.gnome.org/extension/615/appindicator-support/)
* Lock the screen
* AppIndicator disappears

**Comment by drivera73** [comment]
This also applies to Ubuntu 19.04 with GNOME 3.32.0, using both the version from this repository as well as the one from the ubuntu-appindicators package.

Here's a [commit for TopIcons](https://github.com/zhangkaizhao/gnome-shell-extension-tray-icons/commit/95916cab86ef7ae42809ff92386548129617038b) which I stumbled upon while researchign this issue. I know it's a different extension, but there might be useful information in there.

They implemented their fix by latching on to imports.misc.gnomeSession and listening for active/idle events.

**Comment by wuchyi** [comment]
Yup, experincing the same issue on Pop!_OS 19.04, Gnome 3.32.0

**Comment by wuchyi** [comment]
Stiill getting the same issue in 3.32.1

Is this a Gnome shell issue as experienced by TopIcons?

**Comment by drivera73** [comment]
It may well be that Gnome-shell changed the way idle was handled and that's why TopIcons had to incorporate a specific fix for it.  I'm not knowledgeable enough on extension development to adapt the fix to this extension.  I'll try again today and see if I can come up with a workable patch.

Cheers!

**Comment by drivera73** [comment]
After some poking around, I found the following warning in the gnome-shell output:

`[AppIndicatorSupport-WARN] Failed to acquire org.kde.StatusNotifierWatcher`

The direct impact of this can be felt in line 50 of the extension:

`
if (isEnabled && !watchDog.isPresent && statusNotifierWatcher === null)
`
Basically, the **_watchDog_** is an object that's created to interact with **_org.kde.StatusNotifierWatcher_**.  If that name isn't available (as the warning suggests), then _**watchDog.isPresent**_ will never, ever be false.  So the extension will never be rendered.

The correct approach here would be to somehow track if the _**org.kde.StatusNotifierWatcher**_ name was successfully acquired.  The warning in the log suggests that it wasn't, and yet by debugging the code I was able to determine that the code does, indeed, proceed as if it was.  So there's no way to determine if it was or wasn't, and therefore there's no way to determine if the watchDog's isPresent flag should be given any weight or not.

In my case, I've modified the locally-installed extension to not look at that variable, but this isn't a robust solution.  The true solution would be to somehow cleanly detect if we were able to acquire the StatusNotifierWatcher or not.

Doing some more digging I came about [this other issue for this extension](https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/107) and it seems to be related to _**indicator-application-service**_. Then again, this is from more than a year ago so I'm not positive uninstalling that service is the solution nowadays.

I'll keep digging...maybe some other solution will present itself...
